labstep/__init__.py

```diff
@@ -1,581 +1,646 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+
 import requests
 import json
+from datetime import datetime
+
+API_ROOT='https://api.labstep.com/'
 
+def url_join(*args):
+    '''
+    Join a set of args with a slash (/) between them. Instead of
+    checking for the presence of a slash, the slashes from both sides
+    of each arg will be .strip() and then args .join() together.
+    '''
+    return "/".join(arg.strip("/") for arg in args)
 
 
 def login(username,password):
-  '''
-  Returns an authenticated labstep User object to allow you to interact with the labstep API.
+    '''
+    Returns an authenticated Labstep User object to allow 
+    you to interact with the Labstep API.
   
-  Parameters
-  ----------
-  username : str
-    You labstep username
-  password : obj
-    Your labstep password
-
-  Returns
-  -------
-  user
-    An object representing a user on labstep.
-  '''
-  data = {'username': username,
-          'password': password}
-  url = 'https://api.labstep.com/public-api/user/login'
-  r = requests.post(url, json=data, headers={}) 
-  return json.loads(r.content)
+    Parameters
+    ----------
+    username (str)
+        Your Labstep username.
+    password (obj)
+        Your Labstep password.
+
+    Returns
+    -------
+    user
+        An object representing a user on Labstep.
+    '''
+    data = {'username': username,
+            'password': password}
+    url = 'https://api.labstep.com/public-api/user/login'
```

Comment:
use `url_join(API_ROOT,  etc etc)` so that we can change the API_ROOT for everything in one go.